### PR TITLE
adding sample plugins to maven build

### PR DESCRIPTION
**Pull request by tulinkry**
once again and better
#1133 added sample plugins to ant build
this PR should hopefully add the plugins directory into maven build and track it with travis-ci

(https://travis-ci.org/OpenGrok/OpenGrok/builds/145835366#L3581)

### CODE REVIEW

File: plugins/pom.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Copyright (c) 2016, Oracle and/or its affiliates. All rights reserved.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.opensolaris.opengrok</groupId>
+    <artifactId>plugins</artifactId>
+    <packaging>jar</packaging>
+    <version>0.13-SNAPSHOT</version>
```
Comment (tarzanek) [inline plugins/pom.xml]: does this version need to be hardcoded? can it be inherited? or project.version used?

File: plugins/pom.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    <name>OpenGrok Sample Plugins</name>
+
+    <parent>
+        <groupId>org.opensolaris.opengrok</groupId>
+        <artifactId>opengrok</artifactId>
+        <version>0.13-SNAPSHOT</version>
```
Comment (tarzanek) [inline plugins/pom.xml]: dtto as above

File: plugins/pom.xml (hunk 1 above)
Comment (tulinkry) [inline plugins/pom.xml]: This version can be ommited which means it gets inherited. However I unified the style with the other pom files in other submodules where this is always present.

File: plugins/pom.xml (hunk 2 above)
Comment (tulinkry) [inline plugins/pom.xml]: I dug a bit but it seems this cannot be replace by an expression. I've found some workarounds but they don't work in every case.

As above, I got inspired by the other pom files in the opengrok (opengrok-indexer, opengrok-web) where this is not solved.

**Comment by tarzanek** [comment]
will merge for now, however I'd love to have the version just in one pom

**Pull request merged**